ts/features/bonus/cgn/components/merchants/CgnMerchantsListView.tsx

```diff
@@ -1,27 +1,43 @@
 import * as React from "react";
 import { View } from "native-base";
 import { FlatList, ListRenderItemInfo } from "react-native";
-import { TmpMerchantType } from "../../screens/merchants/CgnMerchantsListScreen";
 import { IOStyles } from "../../../../../components/core/variables/IOStyles";
 import ItemSeparatorComponent from "../../../../../components/ItemSeparatorComponent";
 import { EdgeBorderComponent } from "../../../../../components/screens/EdgeBorderComponent";
+import { OfflineMerchant } from "../../../../../../definitions/cgn/merchants/OfflineMerchant";
+import { OnlineMerchant } from "../../../../../../definitions/cgn/merchants/OnlineMerchant";
+import { Merchant } from "../../../../../../definitions/cgn/merchants/Merchant";
 import CgnMerchantListItem from "./CgnMerchantListItem";
 
 type Props = {
-  merchantList: ReadonlyArray<TmpMerchantType>;
-  onItemPress: () => void;
+  merchantList: ReadonlyArray<OfflineMerchant | OnlineMerchant>;
+  onItemPress: (id: Merchant["id"]) => void;
 };
 
 // Component that renders the list of merchants as a FlatList
 const CgnMerchantsListView: React.FunctionComponent<Props> = (props: Props) => {
-  const renderListItem = (listItem: ListRenderItemInfo<TmpMerchantType>) => (
-    <CgnMerchantListItem
-      category={listItem.item.category}
-      name={listItem.item.name}
-      location={listItem.item.location}
-      onPress={props.onItemPress}
-    />
-  );
+  const renderListItem = (
+    listItem: ListRenderItemInfo<OfflineMerchant | OnlineMerchant>
+  ) => {
+    if (OfflineMerchant.is(listItem.item)) {
+      return (
+        <CgnMerchantListItem
+          category={listItem.item.productCategories[0]}
+          name={listItem.item.name}
+          location={listItem.item.address.full_address}
+          onPress={() => props.onItemPress(listItem.item.id)}
+        />
+      );
+    }
+    return (
+      <CgnMerchantListItem
+        category={listItem.item.productCategories[0]}
+        name={listItem.item.name}
+        location={listItem.item.websiteUrl}
+        onPress={() => props.onItemPress(listItem.item.id)}
+      />
+    );
```

Comment:
```suggestion
    const location = OfflineMerchant.is(listItem.item)
      ? listItem.item.address.full_address
      : listItem.item.websiteUrl;
    return (
      <CgnMerchantListItem
        category={index(0, [...listItem.item.productCategories]).toUndefined()}
        name={listItem.item.name}
        location={location}
        onPress={() => props.onItemPress(listItem.item.id)}
      />
    );
```
The proposal is about less code + Safeness 🦺 
Could category be `ProductCategoryEnum | undefined` instead of a string?